eng/tsp-core/pipelines/jobs/single-exe.yml

```diff
@@ -0,0 +1,40 @@
+parameters:
+  - name: name
+    type: string
+  - name: pool
+    type: string
+  - name: vmImage
+    type: string
+  - name: imageKey
+    type: string
+    default: image
+  - name: os
+    type: string
+
+jobs:
+  - job: single_exe_${{ replace(parameters.name, '-', '_') }}
+    displayName: Build single exe ${{ parameters.name }}
+    pool:
+      name: ${{ parameters.pool }}
+      ${{parameters.imageKey}}: ${{ parameters.vmImage }}
```

Comment:
In the pool. For some really weird reason we needed to specify two sets of inputs.

So for windows/ubuntu, the `pool` would look like:

```toml
pool:
   name: $(linuxpool)
   image: $(linuxvmimage)
   os: ${{ parameter.os }}
```

then for `mac` it would look like:


```toml
pool:
   name: $(linuxpool)
   vmImage: $(linuxvmimage)
   os: ${{ parameter.os }}
```

[reference](https://github.com/Azure/azure-sdk-for-js/blob/d48e49d24dee14bcdf44af8388f0520e4da528a1/eng/pipelines/templates/jobs/ci.tests.yml#L45)

Reply:
In the 2nd example $(macpool) etc.